### DISCUSSION TITLE
Getting image coordinates from edited part of an image using ImageMagik.

### DISCUSSION THREAD

**Discussion by ccathal** [ImageMagick/ImageMagick#3312] [Help]
Hi there, I have images I am using for a research project that I need annotated. For these images I am required to identify columns of an image which have some feature in them. Using magick display I can display and edit the image by drawing over the columns which have the feature displayed. Is there some possible way to extract the coordinates from the image which have been drawn on (I really only need the column coordinates of the image). Thank you in advance for any help.

**Comment by fmw42** [comment]
What OS?  What version of IM?

**Reply by ccathal** [reply]
Apologies, I should have added that.
Version: ImageMagick 7.0.10-22 Q16 x86_64 2020-08-24.
OS: Red Hat Enterprise Linux 8.1 (Ootpa)

**Reply by fmw42** [reply]
I am not sure what you are asking. Do you want to know coordinates for the cursor? If so, then I believe X11 GUI allows you to press certain keys to get the color value and coordinates at the cursor. I do not have that ability on my Mac any longer since it went to XQuartz, so I do not recall. If you want the coordinates of something already drawn, that is not possible unless it is a unique color for one pixel only. Then you can search for the coordinates of a given color.  Or you can find the upper left corner of the exact same text by making it into an image and using magick compare to locate that text. Please clarify further what it is you want or provide some example image.

**Reply by ccathal** [reply]
![image](https://user-images.githubusercontent.com/38156633/109382682-dca57780-78d9-11eb-9cbd-353b5c1e7c97.png)

For example, the above image I have annotated with a red line. I might be worth mentioning the all my images are monochrome so choosing a red line might easily identify the coordinates of that line in contrast to the monochrome images. What I would like as an output is a vector of length width of the image with values 1 or 0. 1 indicates that there is a red line in that column and 0 means absence of a red line in that column. However, just getting the coordinates of all red lines in the image will suffice and I can perform some post-processing after.

**Comment by snibgo** [comment]
Please post actual image files, not files that are images of screenshots of some unknown software showing the image on your screen.

You could convert to HSL, take just the Saturation channel, and scale to a single row. In that row, any non-zero pixels had a colour in the corresponding column.

**Comment by fmw42** [comment]
You can change all colors but red to black and then red to white. Then use txt: to output the image as text values. Then use Unix grep to find all 255 values (white). Then use sed command to parse out only the coordinates for values 255.  If you want all pixels 0 or 255, then remove the grep line.

![bw_red](https://user-images.githubusercontent.com/13458218/109393363-09f02680-78d6-11eb-8461-aac64f0c8fc2.png)

```
convert bw_red.png -alpha off \
-fuzz 20% -fill black +opaque red -fill white -opaque red \
-type bilevel txt: | \
grep 'gray(255)' | \
sed -n 's/^\([^ ]*\) .* gray(\(.*\))$/\1 \2/p'

```


```
60,32: 255
61,32: 255
62,32: 255
63,32: 255
64,32: 255
65,32: 255
66,32: 255
67,32: 255
68,32: 255
69,32: 255
70,32: 255
71,32: 255
72,32: 255
73,32: 255
74,32: 255
75,32: 255
76,32: 255
77,32: 255
78,32: 255
79,32: 255
80,32: 255
81,32: 255
80,33: 255
81,33: 255
82,33: 255
83,33: 255
84,33: 255
85,33: 255
86,33: 255
87,33: 255
88,33: 255
89,33: 255
90,33: 255
91,33: 255
93,35: 255
94,35: 255
95,35: 255
96,35: 255
97,35: 255
98,35: 255
99,35: 255
100,35: 255
101,35: 255
102,35: 255
103,35: 255
104,35: 255
105,35: 255
106,35: 255
107,35: 255
106,36: 255
107,36: 255
108,36: 255
109,36: 255
110,36: 255
111,36: 255
112,36: 255
400,59: 255
401,59: 255
402,59: 255
403,59: 255
404,59: 255
405,59: 255
406,59: 255
407,59: 255
408,59: 255
409,59: 255
410,59: 255
411,59: 255
412,59: 255
413,59: 255
414,59: 255
415,59: 255
416,59: 255
417,59: 255
418,59: 255
419,59: 255
420,59: 255
421,59: 255
422,59: 255
423,59: 255
424,59: 255
425,59: 255
426,59: 255
427,59: 255
428,59: 255
429,59: 255
430,59: 255
431,59: 255
432,59: 255
433,59: 255
434,59: 255
435,59: 255
436,59: 255
437,59: 255
438,59: 255
439,59: 255
440,59: 255
441,59: 255
396,60: 255
397,60: 255
398,60: 255
399,60: 255
400,60: 255
345,62: 255
346,62: 255
347,62: 255
348,62: 255
349,62: 255
350,62: 255
351,62: 255
352,62: 255
353,62: 255
354,62: 255
355,62: 255
356,62: 255
357,62: 255
358,62: 255
359,62: 255
360,62: 255
361,62: 255
362,62: 255
363,62: 255
364,62: 255
365,62: 255
366,62: 255
367,62: 255
368,62: 255
369,62: 255
370,62: 255
371,62: 255
372,62: 255
373,62: 255
374,62: 255
375,62: 255
376,62: 255
377,62: 255
378,62: 255
379,62: 255
380,62: 255
381,62: 255
382,62: 255
383,62: 255
384,62: 255
385,62: 255
386,62: 255
387,62: 255
388,62: 255
389,62: 255
390,62: 255
391,62: 255
392,62: 255
393,62: 255
394,62: 255
333,63: 255
334,63: 255
335,63: 255
336,63: 255
337,63: 255
338,63: 255
339,63: 255
340,63: 255
341,63: 255
342,63: 255
343,63: 255
332,64: 255
333,64: 255
322,65: 255
323,65: 255
324,65: 255
325,65: 255
326,65: 255
327,65: 255
328,65: 255
329,65: 255
330,65: 255
331,65: 255
332,65: 255
333,65: 255
310,66: 255
311,66: 255
312,66: 255
313,66: 255
314,66: 255
315,66: 255
316,66: 255
317,66: 255
318,66: 255
319,66: 255
320,66: 255
321,66: 255
322,66: 255
310,67: 255
311,67: 255
303,68: 255
304,68: 255
305,68: 255
306,68: 255
307,68: 255
308,68: 255
309,68: 255
310,68: 255
```
